libs/api/domains/auth/src/lib/models/delegation.model.ts

```diff
@@ -0,0 +1,70 @@
+import {
+  Field,
+  ID,
+  InterfaceType,
+  ObjectType,
+  registerEnumType,
+} from '@nestjs/graphql'
+import {
+  DelegationProvider,
+  DelegationType,
+} from '@island.is/clients/auth-public-api'
+
+registerEnumType(DelegationProvider, { name: 'AuthDelegationProvider' })
+registerEnumType(DelegationType, { name: 'AuthDelegationType' })
+
+@InterfaceType('AuthDelegation', {
+  resolveType(delegation: Delegation) {
+    switch (delegation.type) {
+      case DelegationType.LegalGuardian:
+        return LegalGuardianDelegation
+      case DelegationType.ProcurationHolder:
+        return ProcuringHolderDelegation
+      case DelegationType.Custom:
+        return CustomDelegation
+    }
+  },
+})
+export abstract class Delegation {
```

Comment:
Would like to see the id fields here as well (which exists on the delegation)

Reply:
We only have an id field for database (custom) delegations. The legal guardian and procuring holder delegations don't have an id, unless we make one up.